### PR TITLE
Extract meaningful methods in join operator

### PR DESCRIPTION
**Pull request by skrzypo987**
## Description
Just a simple refactoring

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by lukasz-stec — APPROVED** [review]
good idea + lgtm

**Comment by skrzypo987** [comment]
> good idea

Just for the record, it was your idea in the first place.

**Review by sopel39 — APPROVED** [review]
Could this have perf impact? Did JMH change?

File: core/trino-main/src/main/java/io/trino/operator/join/DefaultPagesHash.java [hunk 1]
```diff
@@ -88,44 +88,59 @@ public DefaultPagesHash(
...
+    }
 
-        size = sizeOf(addresses.elements()) + pagesHashStrategy.getSizeInBytes() +
-                sizeOf(keys) + sizeOf(positionToHashes);
+    private void insertValue(PositionLinks.FactoryBuilder positionLinks, int realPosition, byte hash, int pos)
+    {
+        // look for an empty slot or a slot containing this key
+        while (keys[pos] != -1) {
+            int currentKey = keys[pos];
+            if (hash == positionToHashes[currentKey] && positionEqualsPositionIgnoreNulls(currentKey, realPosition)) {
```
[the remark this answers is not in the document]
Comment (skrzypo987) [reply core/trino-main/src/main/java/io/trino/operator/join/DefaultPagesHash.java:131]: `hash` is now casted in method invocation, line 122

**Review by skrzypo987 — COMMENTED** [review]
> Could this have perf impact? Did JMH change?

It's quite improbable, for two reasons:
- Those methods are inlined by JIT, if applicable
- Every extracted method either operates on batch of values or performs multiple random memory accesses. So the methods are not cheap and cost of invocation is negligible.

**Pull request merged**